### DISCUSSION TITLE
Editing local JSON model of dashboard & previewing (live render?)

### DISCUSSION THREAD

**Discussion by jacksongoode** [grafana/grafana#47836] [Dashboards, Panels & Visualizations]
I've been doing some recent experiments with Selenium and was wondering if it's actually possible to modify the JSON model that's loaded in the browser's session/cache? I could be getting this wrong but I imagine the JSON is pulled via API initially then the frontend evaluates/renders it all in place.

For my purposes, I would like to edit that model without interacting with the page itself but through a saved cache. This would be pre-post so the changes would be entirely local. I feel like this data is somewhere in the cache but I'm new to this live debugging business.

Any help would be appreciated!

**Comment by jacksongoode** [comment]
I think this would naturally mean one could draft edits to a dashboard and see them rendered in the browser without having to interact with the API (or GUI).

There are a few meaningful cases where this would be incredibly helpful such as being able to make edits to variables to see how they effect a panel without having to go into a new view to change the variable and return to the page to see the changes to the panels. Or simultaneous changes across multiple panels at once.

**Comment by jacksongoode** [comment]
I've been working on a project that might resolve the issue of the workflow of provisioning dashboards from a developer perspective see https://github.com/grafana/grafana/issues/13823 and it shares similarities to this project https://motif.land/blog/syncing-text-files-using-yjs-and-the-file-system-access-api that uses the File Sync Api to sync local changes to files (in Grafana's case we're talking about the JSON model) to the browser's file state.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
